seed/scripts/main.js

```diff
@@ -0,0 +1,166 @@
+import GitHub from 'github-api'
+
+import {tech as TechMapper, project as ProjectMapper} from '../../src/helpers/store.js'
+import techs from '../techs/index.js'
+import projects from '../projects/index.js'
+
+const Logger = {
+  debug (module) {
+    return (err) => {
+      console.log('-------------------')
+      console.log(`${module}: ${err}`)
+      console.log('-------------------')
+    }
+  },
+  log (message){
+    return (data) => {
+      console.log(message)
+      return data
+    }
+  }
+}
+
+const Async = {
+  createPromise (module) {
+    return Promise
+    .resolve()
+    .catch(Logger.debug(module))
+  },
+  createQueue (fns, promise) {
+    return fns.reduce((p, fn) => {
+      return p.then(fn)
+    }, promise)
+  }
+}
+
+const DB = {
+  project: {name: 'project', mapper: ProjectMapper},
+  tech: {name: 'tech', mapper: TechMapper},
+  store (item, db) {
+    return db.mapper
+    .create(item)
+    .catch(`${db.name}Store`)
+  },
+  find (query, db) {
+    return db.mapper
+    .findAll(query)
+    .then(items => items.length > 0 ? items[0] : null)
+    .catch(`${db.name}Find`)
+  }
+}
+
+const GH = {
+  getFullName (link) {
+    const path = link.split('github.com/')[1]
+    const [owner, repo] = path.split('/')
+    if (owner && repo) {
+      return `${owner}/${repo}`
+    }else{
+      throw `Cannot get fullName from: ${link}`
+    }
+  },
+  search (fullName) {
+    const search = new GitHub().search()
+
+    return search
+    ._request('GET', `https://api.github.com/repos/${fullName}`)
+    .catch(Logger.debug('github'))
+    .then(result => result.data)
+  }
+}
+
+const Tech = {
+  createQuery (tech) {
+    return () => {
+      return {where: {id: {'===': tech.id}}}
+    }
+  },
+  createTask (tech, position, total) {
+    const id = tech.id
+
+    return Async
+    .createPromise(tech)
+    .then(Logger.log(`Tech: ${position+1}/${total}`))
+    .then(Logger.log(`${id}: searching on DB`))
+    .then(Tech.createQuery(tech))
+    .then((query) => {
+      return DB.find(query, DB.tech)
+    })
+    .then(savedTech => {
+      if (savedTech) {
+        throw 'is already saved'
+      }
+    })
+    .then(Logger.log('saving on DB'))
+    .then(() => {
+      DB.store(tech, DB.tech)
+    })
+    .then(Logger.log('saved'))
+    .catch(Logger.debug(id))
+  }
+}
+
+const Project = {
+  createQuery (fullName) {
+    return {where: {full_name: {'===': fullName}}}
+  },
+  createTask (project, position, total) {
+    let fullName = ''
+
+    return Async
+    .createPromise()
+    .then(() => {
+      fullName = GH.getFullName(project.link)
+    })
+    .then(Logger.log(`Project: ${position+1}/${total}`))
+    .then(Logger.log('searching on DB'))
+    .then(Project.createQuery(fullName))
+    .then((query) => {
+      return DB.find(query, DB.project)
```

Comment:
That's why DB shouldn't know about project or tech. You use one method from DB and use one object that DB already knows as parameter. If you would still leave project inside DB, would be better to have a method `findProject(query)` but even that it's not quite right. Because you get the query also from Project. If Project had every information necessary inside, query and DB.project in that case, you could pass directly to DB.find. Like `DB.find(Project)` or `DB.find(Tech)`. Both belong to the same kind(class) of object, let's call DatabaseItem for example. Which would make it easy for Project and Tech to extend from. Improving the DB API, because now it would expect a DatabaseItem kind of class to be able to exist.